lucet-runtime/lucet-runtime-internals/src/instance/signals.rs

```diff
@@ -245,12 +247,29 @@ extern "C" fn handle_signal(signum: c_int, siginfo_ptr: *mut siginfo_t, ucontext
                 true
             }
         };
-
+        // If we are switching to the host context and we are not processing a SIGALRM from a
+        // KillSwitch, then we were either externally signalled in some other manner, or we faulted
+        // during execution in a guest region.
+        //
+        // In either case, we will need to unlock and update the execution domain ourselves.
         if switch_to_host {
-            // we must disable termination so no KillSwitch may fire in host code.
-            inst.kill_state.disable_termination();
+            // This debug assert exists to make sure that this is not a blocking operation.
+            debug_assert_eq!(inst.kill_state.domain_is_locked(), false);
```

Comment:
A `KillSwitch` could fire exactly after this assert, but before caring about `execution_domain`, so it's not exactly correct :grimacing: